File: examples/quickstart.py
"""Quickstart: run a multi-agent debate with zero API keys."""

import asyncio
import sys
from pathlib import Path

try:
    from aragora_debate import Arena, StyledMockAgent
except ModuleNotFoundError:
    debate_src = Path(__file__).resolve().parents[1] / "aragora-debate" / "src"
    if debate_src.is_dir():
        sys.path.insert(0, str(debate_src))
    from aragora_debate import Arena, StyledMockAgent

agents = [
    StyledMockAgent("analyst", style="supportive"),
    StyledMockAgent("critic", style="critical"),
    StyledMockAgent("pm", style="balanced"),
]
arena = Arena(question="Should we migrate to microservices?", agents=agents)
result = asyncio.run(arena.run())
print(result.receipt.to_markdown())

File: examples/quickstart.ts
/**
 * Quickstart: run a multi-agent debate via the Aragora API.
 *
 * Usage:
 *   npx ts-node examples/quickstart.ts
 *
 * Environment:
 *   ARAGORA_API_URL - API URL (default: http://localhost:8080)
 *   ARAGORA_API_KEY - Your API key (optional for demo mode)
 */

import { AragoraClient } from "@aragora/sdk";

const client = new AragoraClient({
  baseUrl: process.env.ARAGORA_API_URL || "http://localhost:8080",
  apiKey: process.env.ARAGORA_API_KEY,
});

async function main() {
  const result = await client.debates.create({
    task: "Should we use microservices or a monolith?",
    agents: ["claude", "openai"],
    rounds: 3,
  });
  console.log(result.summary);
}

main().catch(console.error);
